pkg/event/target/nsq.go

```diff
@@ -61,19 +75,45 @@ type NSQTarget struct {
 	id       event.TargetID
 	args     NSQArgs
 	producer *nsq.Producer
+	store    Store
 }
 
 // ID - returns target ID.
 func (target *NSQTarget) ID() event.TargetID {
 	return target.id
 }
 
-// Save - Sends event directly without persisting.
+// Save - saves the events to the store which will be replayed when the nsq connection is active.
 func (target *NSQTarget) Save(eventData event.Event) error {
+	if target.store != nil {
+		return target.store.Put(eventData)
+	}
+	if err := target.producer.Ping(); err != nil {
+		// To treat "connection refused" errors as errNotConnected.
+		if isConnRefusedErr(err) {
+			return errNotConnected
+		}
+		return err
+	}
 	return target.send(eventData)
 }
 
-func (target *NSQTarget) send(eventData event.Event) (err error) {
+// isConnRefusedErr - To check fot "connection refused" error.
+func isConnRefusedErr(err error) bool {
```

Comment:
If this function is being used by multiple files in this package, it can be defined in a common file and used by other notification targets.

Reply:
@kannappanr Since most of the PRs need this and are getting reviewed concurrently, I will refactor this and also docs in one final yet-to-merge PR